racket/collects/racket/match/parse-helper.rkt

```diff
@@ -169,21 +168,9 @@
 (define (match-expander-transform parse expander stx accessor
                                   error-msg)
   (let* ([expander* (syntax-local-value expander)]
-         [transformer (accessor expander*)]
-         ;; this transformer might have been defined w/ `syntax-id-rules'
-         [transformer (if (set!-transformer? transformer)
-                          (set!-transformer-procedure transformer)
-                          transformer)])
+         [transformer (accessor expander*)])
     (unless transformer (raise-syntax-error #f error-msg expander*))
-    (define introducer (make-syntax-introducer))
-    (parameterize ([current-match-introducer introducer])
-      (let* ([mstx (introducer (syntax-local-introduce stx))]
-             [mresult (if (procedure-arity-includes? transformer 2)
```

Comment:
The arity-2 code seems not supported by this change.

Reply:
Thanks, good catch. I’ve added a test case for that situation and pushed a fix.